src/Products/Annotation/Controllers/AnnotationApiController.cs

```diff
@@ -73,357 +56,178 @@ public AnnotationConfiguration LoadConfig()
+
+                    List<string> pagesContent;
+
+                    // TODO: execute only if !loadAllPages
+                    pagesContent = GetAllPagesContent(password, documentGuid, info);
+
+                    for (int i = 0; i < info.PageCount; i++)
                     {
-                        List<PageImage> images = AnnotationImageHandler.GetPages(document, imageOptions);
-                        Stream imageStream = images[0].Stream;
-                        
-                        imageStream.Position = 0;
-                        imageStream.CopyTo(memoryStream);
-                        bytes = memoryStream.ToArray();
-                        foreach (PageImage page in images)
+                        PageDataDescriptionEntity page = new PageDataDescriptionEntity
+                        {
+                            number = i + 1,
+                            height = info.PagesInfo[i].Height,
+                            width = info.PagesInfo[i].Width,
+                        };
+
+                        if (annotations != null && annotations.Length > 0)
+                        {
+                            page.SetAnnotations(AnnotationMapper.instance.MapForPage(annotations, i+1, info.PagesInfo[i]));
+                        }
+
+                        if (pagesContent.Count > 0)
                         {
-                            page.Stream.Close();
+                            page.SetData(pagesContent[i]);
                         }
+                        description.pages.Add(page);
                     }
                 }
-                string encodedImage = Convert.ToBase64String(bytes);
-                loadedPage.SetData(encodedImage);
-                DocumentInfoContainer documentDescription = AnnotationImageHandler.GetDocumentInfo(documentPath, password);
-                loadedPage.height = documentDescription.Pages[pageNumber - 1].Height;
-                loadedPage.width = documentDescription.Pages[pageNumber - 1].Width;                
-                // return loaded page object
-                return Request.CreateResponse(HttpStatusCode.OK, loadedPage);
-            }
-            catch (System.Exception ex)
-            {
-                // set exception message
-                return Request.CreateResponse(HttpStatusCode.InternalServerError, new Resources().GenerateException(ex, password));
             }
+
+            description.guid = documentGuid;
+            // return document description
+            return description;
         }
 
-        /// <summary>
-        /// Download document
-        /// </summary>
-        /// <param name="path">string</param>
-        /// <param name="annotated">bool</param>
-        /// <returns></returns>
-        [HttpGet]
-        [Route("downloadDocument")]
-        public HttpResponseMessage DownloadDocument(string path)
+        private List<string> GetAllPagesContent(string password, string documentGuid, IDocumentInfo pages)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'GetAllPagesContent' a static method.](https://app.codacy.com/gh/groupdocs-annotation/GroupDocs.Annotation-for-.NET-MVC/pullRequest?prid=5372471)